### PR TITLE
DOC: update DataFrame.to_feather docstring

### LINKED ISSUE #29532: Update DataFrame.to_feather() docstring/api documentation and deprecate fname argument.
#### Code Sample, a copy-pastable example if possible

This code works. 

```python
import pandas as pd
import pandas.util.testing as tm
from io import BytesIO

df = pd.DataFrame({"a": [1, 2, 3], "b": list("abc")})

buf = BytesIO()
df.to_feather(fname=buf)
result = pd.read_feather(path=buf)

tm.assert_frame_equal(result, df)
```
#### Problem description

Documentation doesn't mention accepting a buffer.

```
    def to_feather(self, fname):
        """
        Write out the binary feather-format for DataFrames.

        Parameters
        ----------
        fname : str
            String file path.
        """
```
#### Expected Output

change argument name to better reflect types accepted and be consistent with other io methods.

would need to deprecate `fname` argument.

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]

</details>

### PR DESCRIPTION
**Pull request by arw2019**
- [x] closes #29532
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

The issue was that `DataFrame.to_feather` had an `fname` argument which in fact could be either a filename or a buffer. Most of the work to close #29532 was done in #30338 where `fname` argument was deprecated in favor of `path`. Here I update the docstring to let users know that they can use a buffer.

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -2199,8 +2199,10 @@ def to_feather(self, path, **kwargs) -> None:
 
         Parameters
         ----------
-        path : str
-            String file path.
+        path : str or file-like object
+            If a string, it will be used as Root Directory path. By file-like
+            object, we refer to objects with a write() method, such as
+            a file handler (e.g. via builtin open function) or io.BytesIO.
```
Comment (jbrockmendel) [inline pandas/core/frame.py:2205]: do we describe what file-like means elsewhere?  id be ok with leaving that out.

backticks around `io.BytesIO`

Comment (arw2019) [reply pandas/core/frame.py]: we do describe it in other writer methods and in the user guide - I took it out  (fb37d73)

**Comment by arw2019** [comment]
@simonjayhawkins

File: pandas/core/frame.py [hunk 2]
```diff
@@ -2215,8 +2215,8 @@ def to_feather(self, path, **kwargs) -> None:
 
         Parameters
         ----------
-        path : str
-            String file path.
+        path : str or file-like object
```
Comment (jreback) [inline pandas/core/frame.py:2218]: can you also add the type to path: FilePathOrBuffer[AnyStr],

Comment (jreback) [reply pandas/core/frame.py:2218]: can you also update in pandas/io/feather_format.py (to_feather)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/frame.py (hunk 2 above)
Comment (arw2019) [reply pandas/core/frame.py:2225]: Done

**Comment by jbrockmendel** [comment]
LGTM

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @arw2019

**Pull request merged**